tests/Keboola/Utils/SanitizeColumnNameTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Keboola\Utils;
+
+class SanitizeColumnNameTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider namesToSanitize
+     **/
+    public function testSanitizeColumnName($nameToSanitize, $sanitizedName)
+    {
+        $sanitized = sanitizeColumnName($nameToSanitize);
+        $this->assertEquals($sanitizedName, $sanitized);
+    }
+
+    public function namesToSanitize()
+    {
+        return [
+            [
+                "_~dlažební  %_kostky_~",
+                "dlazebni__kostky"
+            ],[
+                "test-vn-đá cuội",
```

Comment:
This does not seem to be a correct behavior IMO
If you put "test-*" it should output "test-" (if rest of the characters are invalid), no?
dtto with the jp.

Reply:
a simple test suite that shows some of the caveats is here - https://github.com/keboola/kbc-ui/blob/master/src/scripts/utils/string.spec.js